File: EMS_BACKEND1/src/main/java/com/ems/service/AuditLogService.java
package com.ems.service;

import com.ems.dto.AuditLogDTO;

import java.time.LocalDateTime;
import java.util.List;

public interface AuditLogService {
    AuditLogDTO createAuditLog(AuditLogDTO auditLogDTO);
    AuditLogDTO getAuditLogById(Long logId);
    List<AuditLogDTO> getAllAuditLogs();
    List<AuditLogDTO> getAuditLogsByUserId(Long userId);
    List<AuditLogDTO> getAuditLogsByUsername(String username);
    List<AuditLogDTO> getAuditLogsByAction(String action);
    List<AuditLogDTO> getAuditLogsByEntityTypeAndEntityId(String entityType, Long entityId);
    List<AuditLogDTO> getAuditLogsByTimestampBetween(LocalDateTime start, LocalDateTime end);
}

File: employee-management-frontend/src/services/auditLogService.js
// src/services/auditLogService.js
import axiosInstance from "../api/axiosConfig";

const AUDIT_LOG_API_URL = "/audit-logs";

const AuditLogService = {
  getAllAuditLogs: () => {
    return axiosInstance.get(AUDIT_LOG_API_URL);
  },

  getAuditLogById: (id) => {
    return axiosInstance.get(`${AUDIT_LOG_API_URL}/${id}`);
  },

  getAuditLogsByUserId: (userId) => {
    return axiosInstance.get(`${AUDIT_LOG_API_URL}/user/${userId}`);
  },

  getAuditLogsByUsername: (username) => {
    return axiosInstance.get(`${AUDIT_LOG_API_URL}/username/${username}`);
  },

  getAuditLogsByAction: (action) => {
    return axiosInstance.get(`${AUDIT_LOG_API_URL}/action/${action}`);
  },

  getAuditLogsByEntityTypeAndEntityId: (entityType, entityId) => {
    return axiosInstance.get(
      `${AUDIT_LOG_API_URL}/entity/${entityType}/${entityId}`
    );
  },

  getAuditLogsByTimestampBetween: (start, end) => {
    return axiosInstance.get(`${AUDIT_LOG_API_URL}/time-range`, {
      params: { start, end },
    });
  },
};

export default AuditLogService;
